Civi/RemoteTools/JsonForms/FormSpec/Factory/Control/RadiosControlFactory.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/*
+ * Copyright (C) 2024 SYSTOPIA GmbH
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published by
+ *  the Free Software Foundation in version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+declare(strict_types = 1);
+
+namespace Civi\RemoteTools\JsonForms\FormSpec\Factory\Control;
+
+use Civi\RemoteTools\Form\FormSpec\AbstractFormInput;
+use Civi\RemoteTools\JsonForms\FormSpec\ElementUiSchemaFactoryInterface;
+use Civi\RemoteTools\JsonForms\JsonFormsControl;
+use Civi\RemoteTools\JsonForms\JsonFormsElement;
+
+final class RadiosControlFactory extends AbstractControlFactory {
+
+  public function createInputSchema(
+    AbstractFormInput $input,
+    ElementUiSchemaFactoryInterface $factory
+  ): JsonFormsElement {
+    return new JsonFormsControl(
+      $this->getScope($input), $input->getLabel(), $input->getDescription(), ['format' => 'radio']
+    );
+  }
+
+  protected function supportsInput(AbstractFormInput $input): bool {
+    return 'field' === $input->getType() && 'radio' === $input->getInputType();
```

Comment:
Just wondering whether an `is_a()` check would be reasonable instead of validating against the *JSON Forms* property `radio` (or if that should be a constant on `RadiosField`) …

Reply:
It would be possible to use `instanceof`/`is_a()` here. The idea was to be independent of a concrete class so it could be used e.g. for other classes extending `AbstractFormInput` that need the same JSON Forms Control, but different JSON Schema.